api/src/main/java/com/inrupt/client/Response.java

```diff
@@ -152,6 +154,54 @@ public static BodyHandler<Void> discarding() {
             return responseInfo -> null;
         }
 
+        /**
+         * Throws on HTTP error using the provided mapper, or apply the provided body handler.
+         * @param handler the body handler to apply on non-error HTTP responses
+         * @param isSuccess a callback determining error cases
+         * @param exceptionMapper the exception mapper
+         * @return the body handler
+         * @param <T> the type of the body handler
+         */
+        public static <T> Response.BodyHandler<T> throwOnError(
+                final Response.BodyHandler<T> handler,
+                final Function<Response.ResponseInfo, Boolean> isSuccess,
+                final Function<Response.ResponseInfo, ClientHttpException> exceptionMapper
+        ) {
+            return responseinfo -> {
+                if (!isSuccess.apply(responseinfo)) {
+                    throw exceptionMapper.apply(responseinfo);
+                }
+                return handler.apply(responseinfo);
+            };
+        }
+
+        /**
+         * Throws on HTTP error, or apply the provided body handler.
+         * @param handler the body handler to apply on non-error HTTP responses
+         * @param isSuccess a callback determining error cases
+         * @return the body handler
+         * @param <T> the type of the body handler
+         */
+        public static <T> Response.BodyHandler<T> throwOnError(
```

Comment:
This handler can be used in the `jena` and `rdf4j` modules, where we want to throw a generic HTTP exception.